switch to async streaming download:

ikreymer opened the pull request:
- download via presigned URLs via aiohttp instead of boto APIs
- use async methods from stream-zip to generate zip: note that stream-zip still does a sync->async conversion under the hood
- follow-up to #1933 for streaming download improvements
- fixes datapackage.json in multi-wacz to contain the same resources objects with: `name`, `path`, `hash`, `bytes` to match single WACZ. 
- Begin adding additional metadata to multi-wacz files, including `type` (`crawl`, `upload`, `collection`, `qaRun`), `id` (unique id for the object), `title` / `description` if available (for crawl/upload/collection), and `crawlId` for `qaRun`

ikreymer committed:
switch to async streaming download:
- remove unused sync functions
- use async methods from stream-zip
- note that stream-zip still does a sync->async conversion under the hood
- follow-up to #1933 for streaming download improvements

tw4l commented:
When the multi-WACZs being produced in this branch are loaded into ReplayWeb.page, no seed pages or resources are listed. There may be something slightly off, investigating further.

ikreymer committed:
fix path == name in datapackage.json, to match previous behavior
fully remove boto
tests: update test to ensure multi-wacz resources name == path

ikreymer committed:
fix multi-wacz datapackage.json:
only include name, path, hash, bytes in each resource entry!

ikreymer committed:
update test

ikreymer committed:
fix missing import

ikreymer commented:
Should be fixed now! Turns out the datapackage.json was not quite valid, had incorrect `path` in `resources`, not returning equal to `name`, and matching properties to single WACZ!

tw4l commented:
Tested on dev and working well! Nice job

ikreymer committed:
switch back to mostly sync download with requests, but simplified to not require boto

ikreymer committed:
add additional metadata for multi-wacz to be conformant with datapackage v1:
- include 'id', but also 'title', 'description' and 'organization' fields, as well as 'crawlId' when possible
- add 'type' indicating 'crawl', 'upload', 'collection', 'qa-run'

ikreymer committed:
add 'software' to metadata

tw4l commented on backend/btrixcloud/crawls.py:
```diff
@@ -1034,7 +1034,16 @@ async def download_qa_run_as_single_wacz(
+        metadata = {
+            "type": "qa_run",
+            "id": qa_run_id,
+            "crawlId": crawl_id,
```
I might suggest we stick to one of camelcase or snake case here instead of mixing (e.g. `qa_run` and `crawlId`)

ikreymer replied:
will go with `qaRun`

ikreymer committed:
qa_run -> qaRun
